ta4j-core/src/main/java/org/ta4j/core/execution/trade/TradeOnOpenModel.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2023 Ta4j Organization & respective
+ * authors (see AUTHORS)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.execution.trade;
+
+import org.ta4j.core.BarSeries;
+import org.ta4j.core.TradingRecord;
+import org.ta4j.core.num.Num;
+
+/**
+ * An execution model for {@link BarSeriesManager} objects.
+ *
+ * Executes trades on the next bar at the open price.
+ * 
+ * This is used for strategies that explicitly trade just after a new bar opens
+ * at bar index `t + 1`, in order to execute new or close existing trades as
+ * close as possible to the opening price.
+ */
+public class TradeOnOpenModel implements TradeExecutionModel {
```

Comment:
I'm unsure why this behaviour was implemented as it did more than what a user would it would do.

However, I'll re-implement that feature by closing at the last closing price, if the strategy says to close.
```
                if (strategy.shouldOperate(i, tradingRecord)) {
                    // if this is the very last bar in the series
                    **if(i == barSeries.getEndIndex()) {
                        // close it at the last available price
                        tradingRecord.operate(i, barSeries.getBar(i).getClosePrice(), amount);
                    }** else {
                        tradeExecutionModel.apply(i, tradingRecord, barSeries, amount);
                    }
```